README.md

```diff
@@ -1,116 +1,219 @@
+### System requirements
+The `build-all.sh` Script requires docker, bash 4.2+ and gnu-getopt.
 
+To  build  arm64  packages for  *Red  Hat*  or  *Centos*  `qemu` is  required  and
+qemu-docker integration.
 
-2. Run docker image:
-```
-$ ./build-env/bin/docker-run.sh <image name>
+To enable qemu-docker integration on Linux `binfmt` has to be enabled using this
+image before using arm64 compilation:
+```bash
+docker run --rm --privileged docker/binfmt:a7996909642ee92942dcd6cff44b9b95f08dad64
 ```
 
-For example:
-```
-$ ./build-env/bin/docker-run.sh pelion-bionic-source
+The `binfmt` has to be executed once after every reboot.
+
+Build on MacOS is not tested.
+
+### Red Hat Enterprise Linux (RHEL) subscription
+RHEL  images  does  not  include  subscription  credentials.  `RH_USERNAME`  and
+`RH_PASSWORD` variables  with your  Red Hat  subscription credentials  should be
+exported as environment variables before running build scripts:
+
+```bash
+export RH_USERNAME="Your Red Hat username"
+export RH_PASSWORD="Your Red Hat password"
 ```
 
-The root of this repo is mounted to `/pelion-build`.
+Variables will be used to create docker image. If any of the variables would not
+be set, scripts  will interactively ask for them. Credentials  will be stored in
+docker image. Note that for free development subscription only 16 systems can be
+registered (registered  systems can be  removed on RHEL  subscription management
+page).
 
-Please note that `docker-run.sh` script does not use `PELION_DOCKER_PREFIX`.
+To create Red Hat account visit: https://www.redhat.com/wapps/ugc/register.html
 
-### Build environment for debian-10
 
-1. Scripts for creating clean build docker images:
-```
-$ ./build-env/bin/docker-debian-buster-create.sh # Debian 10
+## Build environment
+
+To access  build environment console `docker-run-env.sh`  script was introduced.
+The script accepts options until first positional argument (which is environment
+name).
+
+Example usage:
+```bash
+# run new rhel/8 image for host arch:
+./build-env/bin/docker-run-env.sh rhel
+
+# run centos container (attach to existing)
+./build-env/bin/docker-run-env.sh -c centos
+
+# run 'ls' in centos container
+./build-env/bin/docker-run-env.sh -c centos ls
+
+# run fresh container (and allow later attaching to it)
+./build-env/bin/docker-run-env.sh -c clean centos ls
+
+# recreate docker image and container and run shell in new container
+./build-env/bin/docker-run-env.sh -r -c clean centos
+
+# run new arm64 container
+./build-env/bin/docker-run-env.sh -c=clean -a arm64 centos
 ```
 
-2. Run docker image:
+The  above script  also creates  docker images  with build  essentials and  with
+additional packages  required for generating  source packages (for  example git,
+npm, python) depending on selected environment.
+
+To  add prefix  to docker  images and  containers export  `PELION_DOCKER_PREFIX`
+variable in your shell with your prefix:
+```
+export PELION_DOCKER_PREFIX=${USER}-
```

Comment:
Did we fix [the dash problem](https://github.com/PelionIoT/distro-pelion-edge/issues/101)?

Reply:
Yes. Now we do not have to split docker image name to get distribution name as we have it in environment config.

Reply:
Nice!